arangod/Cluster/CreateCollection.cpp

```diff
@@ -172,40 +174,40 @@ bool CreateCollection::first() {
       }
     }
 
-    if (_result.fail()) {
+    if (res.fail()) {
       // If this is TRI_ERROR_ARANGO_DUPLICATE_NAME, then we assume that a previous
       // incarnation of ourselves has already done the work. This can happen, if
       // the timing of phaseOne runs is unfortunate with asynchronous creation of
       // shards.
       // In this case, we do not report an error and do not increase the version
       // number of the shard in `setState` below.
-      if (_result.errorNumber() == TRI_ERROR_ARANGO_DUPLICATE_NAME) {
+      if (res.errorNumber() == TRI_ERROR_ARANGO_DUPLICATE_NAME) {
         LOG_TOPIC("9db9c", INFO, Logger::MAINTENANCE)
         << "local collection " << database << "/" << shard
         << " already found, ignoring...";
-        _result.reset(TRI_ERROR_NO_ERROR);
+        result(TRI_ERROR_NO_ERROR);
         _doNotIncrement = true;
         return false;
       }
       std::stringstream error;
       error << "creating local shard '" << database << "/" << shard << "' for central '"
-            << database << "/" << collection << "' failed: " << _result;
+            << database << "/" << collection << "' failed: " << res;
       LOG_TOPIC("63687", ERR, Logger::MAINTENANCE) << error.str();
 
-      _result.reset(TRI_ERROR_FAILED, error.str());
+      result(TRI_ERROR_FAILED, error.str());
     }
 
   } catch (std::exception const& e) {
     std::stringstream error;
 
     error << "action " << _description << " failed with exception " << e.what();
     LOG_TOPIC("60514", WARN, Logger::MAINTENANCE) << error.str();
-    _result.reset(TRI_ERROR_FAILED, error.str());
+    result(TRI_ERROR_FAILED, error.str());
   }
 
-  if (_result.fail()) {
+  if (res.fail()) {
```

Comment:
This would now miss the error set in the exception handler.

Reply:
correct